test/e2e/windows/memory_limits.go

```diff
@@ -105,31 +103,53 @@ func overrideAllocatableMemoryTest(f *framework.Framework, allocatablePods int)
 		podType = "memory_limit_test_pod"
 	)
 
-	totalAllocatable := getTotalAllocatableMemory(f)
-
-	memValue := totalAllocatable.Value()
-	memPerPod := memValue / int64(allocatablePods)
-	ginkgo.By(fmt.Sprintf("Deploying %d pods with mem limit %v, then one additional pod", allocatablePods, memPerPod))
-
-	// these should all work
-	pods := newMemLimitTestPods(allocatablePods, imageutils.GetPauseImageName(), podType, strconv.FormatInt(memPerPod, 10))
-	f.PodClient().CreateBatch(pods)
-
-	failurePods := newMemLimitTestPods(1, imageutils.GetPauseImageName(), podType, strconv.FormatInt(memPerPod, 10))
-	f.PodClient().Create(failurePods[0])
+	selector := labels.Set{"kubernetes.io/os": "windows"}.AsSelector()
+	nodeList, err := f.ClientSet.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{
+		LabelSelector: selector.String(),
+	})
+	framework.ExpectNoError(err)
 
+	for range nodeList.Items {
+		pod := v1.Pod{
+			Spec: v1.PodSpec{
+				Containers: []v1.Container{
+					{
+						Image: imageutils.GetPauseImageName(),
+					},
+				},
+				NodeSelector: map[string]string{
+					"kubernetes.io/os": "windows",
+				},
+			},
+		}
+		f.PodClient().Create(&pod)
+	}
```

Comment:
@jsturtevant Is this is right method for adding image name, node selector in Pod spec. If yes, Could you please tell about how to find the correct memory to be assigned to each pod.

Reply:
Yes this is how you would create a pod. You will want to capture the result and to verify the pod started:
https://github.com/kubernetes/kubernetes/blob/7dd03c8326858ceb232d38476db3240a15e8494f/test/e2e/windows/dns.go#L61-L62

Add memory limit like this: https://github.com/kubernetes/kubernetes/blob/7dd03c8326858ceb232d38476db3240a15e8494f/test/e2e/windows/cpu_limits.go#L127-L132

Reply:
you can get that by querying the node for allocatable memory see an example in the `getTotalAllocatableMemory` function

Reply:
@jsturtevant 

- I think we have to add only  `v1.ResourceMemory: status.Allocatable[v1.ResourceMemory]` in `Limits: v1.ResourceList`. 

Am I thinking correct here, do we also have to value for add `v1.ResourceCPU`.

Reply:
yes we only need memory in this case

Reply:
@jsturtevant in case of `pod` we can add `v1.ResourceMemory: status.Allocatable[v1.ResourceMemory]` for each `node.status`, but what should be added as memory for `failurePod`.

Reply:
i can be anything > than what is left over in the total Allocatable in the cluster for windows, since we have used it all up.  something like 500MB or 1GB should be good.